Graphics/Slicer/Math/Contour.hs

```diff
@@ -151,13 +151,13 @@ contourContainsContour parent child = odd noIntersections
     getContourLineSegIntersections :: Contour -> LineSeg -> [Point2]
     getContourLineSegIntersections (PointSequence contourPoints) line
       | null contourPoints = []
-      | otherwise = mapMaybe (saneIntersection . lineIntersection line) $ makeLineSegsLooped contourPoints
-    saneIntersection :: Intersection -> Maybe Point2
-    saneIntersection (IntersectsAt p2) = Just p2
-    saneIntersection NoIntersection = Nothing
-    saneIntersection Parallel = Nothing
-    saneIntersection AntiParallel = Nothing
-    saneIntersection (Colinear _ _) = Nothing
+      | otherwise = mapMaybe (saneIntersection . intersectsWith (Left line) . Left) $ makeLineSegsLooped contourPoints
+    saneIntersection :: Either Intersection PIntersection -> Maybe Point2
```

Comment:
Why did this type change? What's the meaning between left and right?

Reply:
Intersections are linear-algebra intersections. PIntersections are Projective Geometric Algebra intersections.
the type changed by switching to intersectsWith, which now uses PIntersection types when possible, and Intersection types when necessary. Intersection types are only needed when dealing with line segments, instead of ideal lines.

Reply:
Can `PIntersection` take a constructor which wraps `Intersection`? Without more context it's hard to say, but passing around lots and lots of eithers feels smelly as hell.